### DISCUSSION TITLE
expo app and restful api

### DISCUSSION THREAD

**Discussion by chrismatchett** [Koenkk/zigbee2mqtt#10171] [Devices]
Hi,

I am working on an app using Expo (react native). Can I send restful api commands from the app to zigbee2mgtt to get a list of devices and control them?

Many thanks,

Chris

**Comment by Koenkk** [comment]
z2m has an mqtt and websocket api (websocket is used by the frontend and only runs when frontend is enabled through `configuration.yaml`). Api of the frontend is exactly the same as MQTT (https://www.zigbee2mqtt.io/guide/usage/mqtt_topics_and_messages.html#zigbee2mqtt-friendly-name), payload of these websocket message is `{"topic": MQTT_TOPIC, "payload": MQTT_PAYLOAD}`
